diesel_cli/src/infer_schema_internals/sqlite.rs

```diff
@@ -200,10 +200,14 @@ pub fn get_primary_keys(
         format!("PRAGMA TABLE_INFO('{}')", &table.sql_name)
     };
     let results = sql::<pragma_table_info::SqlType>(&query).load::<FullTableInfo>(conn)?;
-    Ok(results
+    let mut collected: Vec<String> = results
         .into_iter()
         .filter_map(|i| if i.primary_key { Some(i.name) } else { None })
-        .collect())
+        .collect();
+    if collected.is_empty() {
```

Comment:
I would like to see a comment here explaining that all sqlite tables either have a rowid or a primary key with a reference to the documentation.

Also we should likely check for the following cases: There is no other column named rowid for the table (if that's not the case fall back to the other variants shown in the documentation). Also that case likely should be covered with some additional test